systemtest/src/test/java/io/strimzi/systemtest/KafkaClusterTest.java

```diff
@@ -143,7 +159,13 @@ public void testKafkaScaleUpScaleDown() {
         for (int brokerId = 0; brokerId < scaleTo; brokerId++) {
             assertTrue(versions, versions.indexOf("(id: " + brokerId + " rack: ") >= 0);
         }
-        // TODO Check for k8s events, logs for errors
+
+        //Test that the new pod does not have errors or failures in events
+        List<Event> events = getEvents("Pod", newPodName);
+        assertThat(events, hasReasons(Scheduled, Pulling, Pulled, Created, Started));
+        assertThat(events, not(hasReasons(Failed, Unhealthy, FailedSync, FailedValidation)));
```

Comment:
`not(hasReasons(...))` is checking that `events` *does not have **all** of* those reasons, which is not the same as checking that it has *none of* those reasons. I think you probably need two matchers: `hasAllOf()` and `hasNoneOf()`